src/dune_rules/artifacts_obj.mli

```diff
@@ -7,8 +7,8 @@ val empty : t
 val make
   :  dir:Path.Build.t
   -> lib_config:Lib_config.t Memo.t
-  -> libs:(Library.t * _ * Modules.t * Path.Build.t Obj_dir.t) list
-  -> exes:(_ * _ * Modules.t * Path.Build.t Obj_dir.t) list
+  -> libs:(Library.t * _ * Modules.t * Path.Build.t Obj_dir.t * Toggle.t) list
+  -> exes:(_ * _ * Modules.t * Path.Build.t Obj_dir.t * Toggle.t) list
```

Comment:
It seems like you pass the toggle to artifacts_obj but you don't do anything with it, any reason for that?

Reply:
I used it originally, but I guess I can remove it now and just pass the right list of libs, exes and emits directly from the caller, if that's the better approach.

Reply:
I'm not sure if it's the better approach, so I'm looking at the code right to find out. I'll list my thought process below.

First, I need to check where are the callers of `Artifacts_obj` that use a library name. Looks like there's only one caller of:

```
val lookup_library : t -> Lib_name.t -> Lib_info.local option
```

That lives in Expander:

```
    (match Artifacts_obj.lookup_library artifacts name with
     | None -> does_not_exist ~what:"Library" (Lib_name.to_string name)
     | Some lib ->
       Mode.Dict.get (Lib_info.archives lib) mode
       |> Action_builder.List.map ~f:(fun fn ->
         let fn = Path.build fn in
         let+ () = Action_builder.path fn in
         Value.Path fn))
```

So now I think about what's going to happen if we pretend that a disabled library does not exist. Since this code is part of the expander, to reproduce this behavior I'd run the following command `dune build %{cma:dir/disabled-lib}`.

From the perspective of the user, dune telling me this is disabled would be rather undesirable. So we need to improve the situation somehow. To do so, we need `Artifacts_obj.lookup_library` to propagate the library being disabled.

I make the conclusion that your decision is indeed correct and we need to pass enabled/disabled to `Artifacts_obj`

Hope that helps.

Reply:
Just to be clear, this process would help on the eventual case that in some future Dune sets "fake" rules for all the disabled libraries in the current context right? As we confirmed the other day that there's no need to set rules for disabled libs right now.

I also wonder, in this hypothetic future where there are fake rules for disabled libraries (for better user errors), how would we deal with the case where two libraries are defined in the same folder with the same name, but one defined on each context? I assume we would skip the creation of fake rules in that case?

Reply:
Yes, we discussed using fake rules to make the error messages better. And indeed the decision was to forego them for now, as it's quite complex to implement. However, there's other cases where we can improve the error messages without much complexity. Expanding percent forms like we have here is an obvious candidate.

In your hypothetical case we would skip the fake rules. As they would overlap with actual rules.

Reply:
I added a test case for this in https://github.com/ocaml/dune/pull/10245

Looks like we don't handle disabled libraries correctly

Reply:
The change requests didn't specify if this is something that should be fixed in this PR. Is that the case? In any case, the PR doesn't seem to change the current behavior.

Reply:
No, there's no need to fix it here.